packages/main/src/Menu.js

```diff
@@ -0,0 +1,490 @@
+ * Note: if the text ditrection is set to Right-to-left (RTL), <code>Arrow Right</code> and <code>Arrow Left</code> functionality is swapped.
+ * <br>
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/Menu.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.Menu
+ * @extends UI5Element
+ * @tagname ui5-menu
+ * @appenddocs MenuItem
+ * @since 1.2.2
+ * @public
+ */
+class Menu extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get staticAreaStyles() {
+		return staticAreaMenuCss;
+	}
+
+	static get staticAreaTemplate() {
+		return staticAreaMenuTemplate;
+	}
+
+	static get dependencies() {
+		return [
+			ResponsivePopover,
+			Button,
+			List,
+			StandardListItem,
+			Icon,
+		];
+	}
+
+	static async onDefine() {
+		Menu.i18nBundle = await getI18nBundle("@ui5/webcomponents");
+	}
+
+	get itemsWithChildren() {
+		return !!this._currentItems.filter(item => item.item.items.length).length;
+	}
+
+	get itemsWithIcon() {
+		return !!this._currentItems.filter(item => item.item.icon !== "").length;
+	}
+
+	get isRtl() {
+		return this.effectiveDir === "rtl";
+	}
+
+	get placementType() {
+		const placement = this.isRtl ? "Left" : "Right";
+		return this._isSubMenu ? placement : "Bottom";
+	}
+
+	get verticalAlign() {
+		return this._isSubMenu ? "Top" : "Bottom";
+	}
+
+	get labelBack() {
+		return Menu.i18nBundle.getText(MENU_BACK_BUTTON_ARIA_LABEL);
+	}
+
+	get labelClose() {
+		return Menu.i18nBundle.getText(MENU_CLOSE_BUTTON_ARIA_LABEL);
+	}
+
+	get isPhone() {
+		return isPhone();
+	}
+
+	get isSubMenuOpened() {
+		return !!this._parentMenuItem;
+	}
+
+	get menuTitlePhone() {
+		return this.isSubMenuOpened ? this._parentMenuItem.text : this.title;
+	}
+
+	onBeforeRendering() {
+		!isPhone() && this._prepareCurrentItems(this.items);
+
+		const itemsWithChildren = this.itemsWithChildren;
+		const itemsWithIcon = this.itemsWithIcon;
+
+		this._currentItems.forEach(item => {
+			item.item._siblingsWithChildren = itemsWithChildren;
+			item.item._siblingsWithIcon = itemsWithIcon;
+		});
+	}
+
+	/**
+	 * Shows the Menu near the opener element.
+	 * @param {HTMLElement} opener the element that the popover is shown at
+	 * @public
+	 */
+	async showAt(opener) {
+		if (isPhone()) {
+			this._prepareCurrentItems(this.items);
+			this._parentItemsStack = [];
+		}
+		if (!this._isSubMenu) {
+			this._parentMenuItem = undefined;
+		}
+		(await this._getPopover()).showAt(opener);
+	}
+
+	/**
+	 * Closes the Menu.
+	 * @public
+	 */
+	close() {
+		if (Object.keys(this._popover).length) {
+			if (isPhone()) {
+				this._parentItemsStack = [];
+			}
+			this._popover.close();
+			this._popover.resetFocus();
+		}
+	}
+
+	async _getPopover() {
+		const id = this._id;
+		this._popover = (await this.getStaticAreaItemDomRef()).querySelector("[ui5-responsive-popover]");
+		this._popover.initialFocus = `${id}-menu-item-0`;
```

Comment:
This can be easily break if the first item is disabled, the focus should go on the first not-disabled item, which might not be always the first item in general.

Reply:
You're right, refactored.